library/std/src/os/fd/owned.rs

```diff
@@ -176,7 +176,20 @@ impl Drop for OwnedFd {
             // something like EINTR), we might close another valid file descriptor
             // opened after we closed ours.
             #[cfg(not(target_os = "hermit"))]
-            let _ = libc::close(self.fd);
+            {
+                use crate::sys::os::errno;
+                // ideally this would use assert_unsafe_precondition!, but that's only in core
+                if cfg!(debug_assertions) {
```

Comment:
it should be fine to make the macro public

Reply:
CC @saethlin can we? I think I saw something about the crates being special-cased somewhere for performance reasons.

Reply:
It should be fine to make it public. I just try to avoid making things public because then I have to deal with the stability attributes.

Reply:
Ah the macro needs const anyway, I'll have to replicate it to be able to call libc functions.